package-res/resources/web/pentaho/type/complex.js

```diff
@@ -616,14 +616,22 @@ define([
         var spec = {};
 
         if(requireType) {
-          // returns the id of the type when it has one, or the actual spec if not...
+          spec._ = this.type.id;
+        }
+
+        if(keyArgs.preferPropertyArray) {
```

Comment:
When preferPropertyArray is honored and true, note that the whole spec is an array. Not an object with a d property that contains an array (like with the list format :-) )